ecl/hql/hqlcache.cpp

```diff
@@ -360,6 +497,23 @@ static IHqlExpression * createSimplifiedDefinitionFromType(ITypeInfo * type)
         {
             return createValue(no_simplified, LINK(type));
         }
+    case type_table:
+        {
+            OwnedHqlExpr record;
+            try
+            {
+                HqlSimplifiedRecordTransformer simplifiedRecordTransformer;
+                record.setown(simplifiedRecordTransformer.createTransformed(queryOriginalRecord(type)));
+            }
+            catch (IException *e)
+            {
+                e->Release();
+                return nullptr;
+            }
+            if (!record || getNumChildTables(record)>0)
```

Comment:
with the final code record cannot be null.
The check for getNumChildTables() could be done before the record is transformed - avoiding the transform.

Reply:
Please add a comment why child tables are not allowed...